### PR TITLE
fix pod termination issue when kubelet mounted as symlink

### PR DESCRIPTION
**Pull request by badri-pathak**
## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## How risky is this change?
- [ ] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by badri-pathak** [comment]
Image: quay.io/badri_pathak/ibm-spectrum-scale-csi-driver:deletion_test5

File: driver/csiplugin/nodeserver.go [hunk 1]
```diff
@@ -293,24 +293,26 @@ func (ns *ScaleNodeServer) NodePublishVolume(ctx context.Context, req *csi.NodeP
 // to be returned if there are any.
 func unmountAndDelete(ctx context.Context, targetPath string, forceful bool) (bool, *csi.NodeUnpublishVolumeResponse, error) {
 	loggerId := utils.GetLoggerId(ctx)
-	klog.V(4).Infof("[%s] unmount and delete %s", loggerId, targetPath)
-	targetPathInContainer := hostDir + targetPath
+	klog.V(4).Infof("[%s] unmount and delete targetPath:[%s], forceful:[%s]", loggerId, targetPath)
```
Comment (deeghuge) [inline driver/csiplugin/nodeserver.go:296]: variable is missing for forceful

**Comment by badri-pathak** [comment]
Image: quay.io/badri_pathak/ibm-spectrum-scale-csi-driver:deletion_test6

File: driver/csiplugin/nodeserver.go (hunk 1 above)
Comment (badri-pathak) [reply driver/csiplugin/nodeserver.go]: Have fixed the log message.

File: driver/csiplugin/nodeserver.go [hunk 2]
```diff
@@ -320,13 +322,15 @@ func unmountAndDelete(ctx context.Context, targetPath string, forceful bool) (bo
 		klog.V(4).Infof("[%s] %v is unmounted successfully", loggerId, targetPath)
 	}
 	// Delete the mount point
-	if err = os.Remove(targetPathInContainer); err != nil {
+	klog.V(4).Infof("[%s] remove mount point ", loggerId)
+	if err = os.Remove(targetPath); err != nil {
+		klog.Errorf("[%s] remove targetPath: [%s] failed with error [%v]", loggerId, targetPath, err)
 		if os.IsNotExist(err) {
-			klog.V(4).Infof("[%s] targetPath [%s] is not present", loggerId, targetPath)
+			klog.V(4).Infof("[%s] inside when targetPath [%s] is not present", loggerId, targetPath)
```
Comment (deeghuge) [inline driver/csiplugin/nodeserver.go:329]: remove the 'inside when'

**Review by deeghuge — CHANGES_REQUESTED** [review]

**Comment by badri-pathak** [comment]
quay.io/badri_pathak/ibm-spectrum-scale-csi-driver:deletion_test7

**Review by deeghuge — APPROVED** [review]

**Review by hemalathagajendran — APPROVED** [review]

**Comment by badri-pathak** [comment]
regression passed: 
image

**Review by Jainbrt — APPROVED** [review]

**Review by saurabhwani5 — APPROVED** [review]

**Pull request merged**